graph/schema.graphqls

```diff
@@ -31,6 +31,7 @@ type FoodRecord {
 enum Sex {
```

Comment:
Currently this value is allowed to be null both in the `users` database table and the `User` GraphQL type. I had imagined that this would allow for the value to remain unset in any case where a user might not choose to identify as male or female (whether because they do not identify with either binary value or simply opt not to share this information).

Do we expect to treat "Other" as a default "undefined" value, or a value meaningful distinct from "null" indicating that the user has indicated that they identify as trans? If our goal is just to enable a user to opt out of a binary label, we can accomplish that with the existing schema with the appropriate UX. For example, something along the lines of:

How do you identify?
* Male (maps to MALE)
* Female (maps to FEMALE)
* Neither or prefer not to say (maps to NULL)

Of course this is also complicated by the fact that gender identity is not necessarily the same as biological sex. Biological sex is the useful signal for us to collect because it is an input into the BMR formula.
    BMR for Men = 66.47 + (13.75 * weight [kg]) + (5.003 * size [cm]) − (6.755 * age [years])
    BMR for Women = 655.1 + (9.563 * weight [kg]) + (1.85 * size [cm]) − (4.676 * age [years])

It's good that we are considering this early. I'd be interested in hearing more in-depth thoughts on what experience we want to provide and how we want to use this personal data that we will collect from our users. Once we have a clear design we can make a better-informed choice about how to change our database. In my opinion the best solution will be the one that is simplest while offering the most inclusive experience.

Reply:
Great points.

With respect to the schema for biological sex, I think the mapping above makes sense.

From a UX perspective we can take two approaches.

We ask the user only for their biological sex with the options:
* Male
* Female
* Neither/prefer not to say.
We use this to calculate BMR (in the null case we can figure out whether we want to default to male or female).

Or we ask the user for their biological sex, and optionally for their gender identity. We use the biological sex to support BMR calculations, and just store the gender identity.

Either way I think our current schema for biological sex is fine. I think we should table whether or not to ask for Gender Identity until we are building out our OOBE.

Reply:
Sounds good, thanks for the follow up.